### PR TITLE
[#950] general oauth2 realization

### PR DESCRIPTION
**Pull request by kirillmakhonin**
This closes #950 
DEX has been removed, it allows Legion to use any oauth2 IP.

### CODE REVIEW

**Review by maxkochubey — APPROVED** [review]

File: terraform/env_types/gcp/gke/k8s_setup/variables.tf [hunk 1]
```diff
@@ -103,47 +103,36 @@ variable "monitoring_namespace" {
   description = "clusterwide monitoring namespace"
```
Comment (mcdoker18) [inline terraform/env_types/gcp/gke/k8s_setup/variables.tf:99]: I think you should delete `github_org_name` variable

File: terraform/env_types/gcp/gke/k8s_setup/main.tf [hunk 2]
```diff
@@ -39,30 +39,18 @@ module "dashboard" {
   tls_secret_crt       = module.base_setup.tls_secret_crt
 }
 
-module "dex" {
```
Comment (mcdoker18) [inline terraform/env_types/gcp/gke/k8s_setup/main.tf:42]: You did not delete the entire dex terraform module.

**Review by mcdoker18 — CHANGES_REQUESTED** [review]

File: terraform/env_types/gcp/gke/k8s_setup/variables.tf (hunk 1 above)
Comment (kirillmakhonin) [reply terraform/env_types/gcp/gke/k8s_setup/variables.tf:99]: Done

File: terraform/env_types/gcp/gke/k8s_setup/main.tf (hunk 2 above)
Comment (kirillmakhonin) [reply terraform/env_types/gcp/gke/k8s_setup/main.tf:42]: Done

**Review by mcdoker18 — APPROVED** [review]

**Review by alexander-semenets — APPROVED** [review]

**Pull request merged**